simuk/sbc.py

```diff
@@ -121,6 +169,14 @@ def _get_posterior_samples(self, prior_predictive_draw):
         posterior = extract(check, group="posterior", keep_dataset=True)
         return posterior
 
+    def _get_posterior_samples_numpyro(self, prior_predictive_draw):
+        """Generate posterior samples using numpyro conditioned to a prior predictive sample."""
+        mcmc = MCMC(self.numpyro_model, **self.sample_kwargs)
+        rng_seed = jax.random.PRNGKey(self._seeds[self._simulations_complete])
+        free_vars_data = {k: v for k, v in self.data_dir.items() if k not in self.observed_vars}
+        mcmc.run(rng_seed, **free_vars_data, **prior_predictive_draw)
+        return from_numpyro(mcmc)["posterior"]
```

Comment:
`from_numpyro` is required to calculate the posterior samples, which means we need to import `arviz`. Is there an alternative way to calculate this using only `arviz_base`?

Reply:
Not yet, but that's something we want in arviz_base

Reply:
Hmm... 
In that case, let's keep `arviz` as a dependency until we have it in `arviz_base`. You can review this PR.